### ISSUE TITLE
400 bad request.  invalid_token.

### ISSUE THREAD

**Issue by levieggertcru** [okta/okta-oidc-ios#328]
Hey I was wondering if you could help shed some light on this issue.  We recently updated our app to use Okta authentication and we are getting a number of our users reporting the same error during authentication.  They're presented with the browser and see 400 bad request.  The id token is invalid message.  This appears to effect only certain users as our internal testers haven't experienced this error.   

**Describe the bug**
Users see 400 bad request.  invalid_token.  

**Screenshots**
![bad_request](https://user-images.githubusercontent.com/59846460/147982535-ee65b6c5-eaea-412d-bb10-24d15958cb7c.png)

**Smartphone (please complete the following information):**
iOS 15.1,

**Comment by levieggertcru** [comment]
I'm going to close this issue as we were able to solve it with our customers.  There were a couple things we did.
1. We added some extra logic to clear any persisted auth sessions on a new app install.  
2. We made some changes to how we initialize our app and fetch data from the Keychain / UserDefaults.  See iOS 15 bug that has been reported here https://developer.apple.com/forums/thread/691353.
